packages/forms/src/UIForm/fields/Datalist/Datalist.component.js

```diff
@@ -74,25 +77,50 @@ class Datalist extends Component {
 	}
 
 	getTitleMap() {
-		const titleMap = this.state.titleMap || this.props.schema.titleMap || [];
+		const isMultiSection = this.props.schema.options && this.props.schema.options.multiSection;
+		const titleMap =
+			this.state.titleMap ||
+			((this.props.schema.options && this.props.schema.options.titleMap) ||
+				this.props.schema.titleMap) ||
+			[];
+		let titleMapFind = titleMap;
 
 		if (!this.props.schema.restricted) {
 			const isMultiple = this.props.schema.schema.type === 'array';
 			const values = isMultiple ? this.props.value : [this.props.value];
+
+			if (isMultiSection) {
+				titleMapFind =
+					titleMap &&
```

Comment:
i think this check should move to the `if(isMultiSection)` conditional and the check should verify that it is an array